packages/cli/src/lib/implementation/config-middleware.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { join } from 'path';
+import { configMiddleware, ConfigParseError } from './config-middleware';
+import { expect } from 'vitest';
+
+const withDirName = (path: string) => join(__dirname, path);
+
+describe('applyConfigMiddleware', () => {
+  it('should load valid config `read-config.mock.mjs`', async () => {
+    const configPathMjs = withDirName('mock/config-middleware-config.mock.mjs');
+    const config = await configMiddleware({ configPath: configPathMjs });
+    expect(config.configPath).toContain('.mjs');
+    expect(config.persist.outputPath).toContain('mjs-');
+  });
+
+  it('should load valid config `read-config.mock.cjs`', async () => {
+    const configPathCjs = withDirName('mock/config-middleware-config.mock.cjs');
+    const config = await configMiddleware({ configPath: configPathCjs });
+    expect(config.configPath).toContain('.cjs');
+    expect(config.persist.outputPath).toContain('cjs-');
+  });
+
+  it('should load valid config `read-config.mock.js`', async () => {
+    const configPathJs = withDirName('mock/config-middleware-config.mock.js');
+    const config = await configMiddleware({ configPath: configPathJs });
+    expect(config.configPath).toContain('.js');
+    expect(config.persist.outputPath).toContain('js-');
+  });
```

Comment:
The filenames in the description don't match what's used in the test. Maybe just mention the file extensions, so it's easier to maintain.

```suggestion
  it('should load valid .mjs config', async () => {
    const configPathMjs = withDirName('mock/config-middleware-config.mock.mjs');
    const config = await configMiddleware({ configPath: configPathMjs });
    expect(config.configPath).toContain('.mjs');
    expect(config.persist.outputPath).toContain('mjs-');
  });

  it('should load valid .cjs config', async () => {
    const configPathCjs = withDirName('mock/config-middleware-config.mock.cjs');
    const config = await configMiddleware({ configPath: configPathCjs });
    expect(config.configPath).toContain('.cjs');
    expect(config.persist.outputPath).toContain('cjs-');
  });

  it('should load valid .js config', async () => {
    const configPathJs = withDirName('mock/config-middleware-config.mock.js');
    const config = await configMiddleware({ configPath: configPathJs });
    expect(config.configPath).toContain('.js');
    expect(config.persist.outputPath).toContain('js-');
  });
```